jax/api.py

```diff
@@ -354,12 +483,64 @@ def xla_computation(fun: Callable,
     ROOT tuple.18 = (f32[], f32[], f32[]) tuple(all-reduce.7, all-reduce.12, all-reduce.17)
   }
   """
+  return _xla_computation(
+      fun,
+      static_argnums=static_argnums,
+      axis_env=axis_env,
+      in_parts=in_parts,
+      out_parts=out_parts,
+      backend=backend,
+      tuple_args=tuple_args,
+      instantiate_const_outputs=instantiate_const_outputs,
+      return_shape=return_shape)
+
+
+_XlaReturn = collections.namedtuple("_XlaReturn", [
+    "xla_computation", "out_shape", "out_pytree_def", "lazy_expressions",
+    "shaped_arrays", "parameter_is_tupled_arguments"
+])
+
+
+def _xla_computation(
+    fun: Callable,
+    static_argnums: Union[int, Iterable[int]] = (),
+    axis_env: Optional[Sequence[Tuple[AxisName, int]]] = None,
+    in_parts=None,
+    out_parts=None,
+    backend: Optional[str] = None,
+    tuple_args: Optional[bool] = None,
+    instantiate_const_outputs: Optional[bool] = True,
+    return_shape: bool = False,
+    xla_return: bool = False,
+    donate_argnums: Union[int, Iterable[int]] = ()
+) -> Callable:
+  """An internal implementation for `xla_computation` and `_cpp_jit`.
+
+  See `xla_computation` for the full documentation.
+
+  Args:
+    tuple_args: If `True`, the resulting XLA computation will have a single
+      tuple argument that is unpacked into the specified function arguments.
+      By default (`None`), this will be enabled when there are more than 100
+      argumments, as it will be more efficient and that XLA do not support
```

Comment:
"By default (`None`), tupling will be enabled when there are more than 100 arguments, since some platforms have limits on argument arity."